### PR TITLE
ENH Faster method for creating injected instances

### PR DESCRIPTION
**Pull request by emteknetnz**
~17% faster method for instantiating things via injector

I validated the performance using the following code in my projects `PageController`

```php
<?php

use SilverStripe\Admin\ModalController;
use SilverStripe\CMS\Controllers\ContentController;

class PageController extends ContentController
{
    protected function init()
    {
        parent::init();

        $time_pre = microtime(true);
        for($i=0;$i<100000;$i++) {
            //$this->injectOld(); // This took 0.2139 seconds 
            $this->injectNew(); // This took 0.1776 seconds 
        }
        $time_post = microtime(true);
        $exec_time = $time_post - $time_pre;
        echo sprintf("This took %.4f seconds", $exec_time);

    }

    private function injectOld()
    {
        $class = ModalController::class;
        $params = ['MyController', 'MyName'];
        $reflector = new ReflectionClass($class);
        return $reflector->newInstanceArgs($params);
    }

    private function injectNew()
    {
        $class = ModalController::class;
        $params = ['MyController', 'MyName'];
        return new $class(...array_values($params));
    }
}

```

### CODE REVIEW

**Review by kinglozzer — APPROVED** [review]
Nice! I can’t see any reason we would be using Reflection here other than old PHP versions (that we no longer have to support)

File: src/Core/Injector/InjectionCreator.php [hunk 1]
```diff
@@ -2,29 +2,16 @@
...
         }
-
-        if (count($params)) {
-            // Remove named keys to ensure that PHP7 and PHP8 interpret these the same way
-            $params = array_values($params);
-            return $reflector->newInstanceArgs($params);
-        }
-
-        return $reflector->newInstance();
+        return new $class(...array_values($params));
```
Comment (GuySartorelli) [inline src/Core/Injector/InjectionCreator.php:15]: Is it necessary to call `array_values` if you're using the spread (`...`) operator? I would have expected that to ignore keys.

If it is necessary, might be worth keeping the comment about keeping things consistent between php7 and 8.

Comment (emteknetnz) [reply src/Core/Injector/InjectionCreator.php:15]: yes

```
~ $ php -a
Interactive mode enabled

php > function f($a,$b){print_r([$a,$b]);};
php > f(...[1,2]);
Array
(
    [0] => 1
    [1] => 2
)
php > f(...['a'=>3,'b'=>4]);
PHP Warning:  Uncaught Error: Cannot unpack array with string keys in php shell code:1
Stack trace:
#0 {main}
  thrown in php shell code on line 1
```

Comment (emteknetnz) [reply src/Core/Injector/InjectionCreator.php:15]: will add back inline comment to explain array values

**Comment by emteknetnz** [comment]
@kinglozzer are you happy to re-approve/merge post small change (it's still functionally the same)

**Review by michalkleiner — APPROVED** [review]
Nice!

File: src/Core/Injector/InjectionCreator.php (hunk 1 above)
Comment (michalkleiner) [reply src/Core/Injector/InjectionCreator.php]: The spread operator works with string keys from PHP 8.1, older versions need to use the array_values approach.

**Pull request merged**